avocado_vt/test.py

```diff
@@ -213,21 +213,25 @@ def verify_background_errors(self):
         Logs all errors in the background_errors into background-error.log and
         error the test.
         """
-        err_file_path = os.path.join(self.logdir, BG_ERR_FILE)
         bg_errors = self.background_errors.get_all()
-        error_messages = ["BACKGROUND ERROR LIST:"]
-        for index, error in enumerate(bg_errors):
-            error_messages.append(
-                "- ERROR #%d -\n%s" % (index, "".join(
-                    traceback.format_exception(*error)
-                    )))
-        genio.write_file(err_file_path, '\n'.join(error_messages))
         if bg_errors:
-            msg = ["Background error"]
-            msg.append("s are" if len(bg_errors) > 1 else " is")
-            msg.append((" detected, please refer to file: "
-                        "'%s' for more details.") % BG_ERR_FILE)
-            self.error(''.join(msg))
+            err_file_path = os.path.join(self.logdir, BG_ERR_FILE)
+            try:
+                error_messages = ["BACKGROUND ERROR LIST:"]
+                for index, error in enumerate(bg_errors):
+                    error_messages.append(
+                        "- ERROR #%d -\n%s" % (index, "".join(
+                            traceback.format_exception(*error)
+                            )))
+                genio.write_file(err_file_path, '\n'.join(error_messages))
+                msg = ["Background error"]
+                msg.append("s are" if len(bg_errors) > 1 else " is")
+                msg.append((" detected, please refer to file: "
+                            "'%s' for more details.") % BG_ERR_FILE)
+                self.error(''.join(msg))
+            except:
+                self.log.warn("Failed to save background error: %s"
```

Comment:
So you want set the test status to be `WARN` in this situation?

Reply:
@luckyh good catch, warning info is enough,  no need to set test status to `WARN`, will update later, thanks !